src/core/gui/dialog/toolbarCustomize/ToolbarDragDropHandler.cpp

```diff
@@ -16,37 +16,28 @@
 
 ToolbarDragDropHandler::ToolbarDragDropHandler(Control* control): control(control) {}
 
-ToolbarDragDropHandler::~ToolbarDragDropHandler() { clearToolbarsFromDragAndDrop(); }
+ToolbarDragDropHandler::~ToolbarDragDropHandler() = default;
 
 void ToolbarDragDropHandler::prepareToolbarsForDragAndDrop() {
-    int len = 0;
     MainWindow* win = control->getWindow();
-    GtkWidget** widgets = win->getToolbarWidgets(len);
+    auto len = win->getToolbarCount();
 
-    this->toolbars = new ToolbarAdapter*[len + 1];
-    this->toolbars[len] = nullptr;
+    this->toolbars.clear();
+    this->toolbars.reserve(len);
 
-    for (int i = 0; i < len; i++) {
-        GtkWidget* w = widgets[i];
-        this->toolbars[i] = new ToolbarAdapter(w, win->getToolbarName(GTK_TOOLBAR(w)),
-                                               control->getWindow()->getToolMenuHandler(), control->getWindow());
+    for (size_t i = 0; i < len; i++) {
+        GtkWidget* w = win->getToolbarWidget(i);
```

Comment:
Can we get a const reference for the `std::array` and iterate on it directly here?

Reply:
Yes, and then we don't have a need for the above `getToolbarCount`